### PR TITLE
Cluster Id

### LINKED ISSUE #7: Generate a cluster id
The first node should generate a cluster id on startup (or first request) and set that in a ccf map. This never changes after initialisation.

### PR DESCRIPTION
**Pull request by jeffa5**
Fixes #7

### CODE REVIEW

**Comment by heidihoward** [comment]
This PR is reusing CCF's service identiy for the member ID, am I understanding this correctly?

**Comment by jeffa5** [comment]
Yes, deriving it from the service certificate. It might not be stable over DR but I think that might be ok for now. We can always rework it later.

**Comment by heidihoward** [comment]
CCF's service certificate should not be stable over DR but I agree that it is fine for now

**Review by heidihoward — APPROVED** [review]
lgtm (you might want to wait for a thumbs up from @jumaffre)

File: src/app/app.cpp [hunk 1]
```diff
@@ -963,11 +977,38 @@ namespace app
...
+
+      if (!service_info.has_value())
+      {
+        // shouldn't but just in case
+        CCF_APP_FAIL("Failed to get id for cluster");
+        return 0;
+      }
+
+      auto cert = service_info.value().cert;
+      auto sha = crypto::sha256(cert.raw());
```
Comment (jumaffre) [inline src/app/app.cpp:1005]: This should be derived from the certificate public key instead, as members[ can renew the service certificate](https://microsoft.github.io/CCF/main/operations/certificates.html#service-certificate), which would change the cluster id. The public key should be extractable via `crypto::make_verifier(cert)->public_key_der()`.

**Review by jumaffre — APPROVED** [review]

**Pull request merged**